File: frontend/passamv/src/app/home/dto/item.ts
import { Option } from "./option";

export class Item {

    public id: number;
    public question: string;
    public module: number;
    public options: Option[];

    public getId() {
        return this.id;
    }

    public setId(id: number) {
        this.id = id;
    }

    public getQuestion() {
        return this.question;
    }

    public setQuestion(question: string) {
        this.question = question;
    }

    public getModule() {
        return this.module;
    }

    public setModule(module: number) {
        this.module = module;
    }

    public getOptions() {
        return this.options;
    }

    public setOptions(options: Option[]) {
        this.options = options;
    }

}

File: backend/financial/financial/src/main/java/com/passamv/financial/entity/Item.java
package com.passamv.financial.entity;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonIdentityReference;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import jakarta.persistence.*;
import lombok.Data;

import java.util.List;

@Data
@Entity(name = "ITEM")
public class Item {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "item_id")
    private Integer id;

    private String question;

    @ManyToOne
    @JoinColumn(name = "module_id", referencedColumnName = "module_id")
    @JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "id")
    @JsonIdentityReference(alwaysAsId = true)
    private Module module;

    @OneToMany(mappedBy = "item")
    private List<Options> options;

    public Item() {}

    public Item(Integer id, String question, Module module) {
        this.id = id;
        this.question = question;
        this.module = module;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getQuestion() {
        return question;
    }

    public void setQuestion(String question) {
        this.question = question;
    }

    public Module getModule() {
        return module;
    }

    public void setModule(Module module) {
        this.module = module;
    }
}
